src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipeManager.java

```diff
@@ -52,6 +65,20 @@ public void addRecipe(@NotNull final CustomRecipe recipe)
         }
 
         recipeMap.get(recipe.getCrafter()).put(recipe.getRecipeId(), recipe);
+
+        if (!recipeOutputMap.containsKey(recipe.getPrimaryOutput().getItem()))
+        {
+            recipeOutputMap.put(recipe.getPrimaryOutput().getItem(), new ArrayList<>());
```

Comment:
How is this going to handle when there are multiple recipes with the same output? We support this today in taught recipes for say charcoal, where you can teach all the logs. If somebody were to add those as a custom recipe, you have a potential problem.